src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -73,10 +73,9 @@ Copyright (c) .NET Foundation. All rights reserved.
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.NET.Windows.targets" Condition="'$(TargetPlatformIdentifier)' == 'Windows'" />
   
   <!-- Import WindowsDesktop targets if necessary -->
-  <PropertyGroup Condition=" '$(ImportWindowsDesktopTargets)' == ''
-                 and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'
-                 and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '3.0'))
-                 and '$(TargetPlatformIdentifier)' == 'Windows'
+  <PropertyGroup Condition=" '$(ImportWindowsDesktopTargets)' == '' and
+                 (('$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '3.0')) and '$(TargetPlatformIdentifier)' == 'Windows')
+                 or '$(TargetFrameworkIdentifier)' == '.NETFramework')
```

Comment:
Instead of complicating the condition with more nested parenthesis, I'd separate out the conditions something like this:

```xml
  <PropertyGroup Condition=" '$(ImportWindowsDesktopTargets)' == ''
                 and ('$(UseWpf)' == 'true' Or '$(UseWindowsForms)' == 'true')">
    <ImportWindowsDesktopTargets Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'
                                 and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '3.0'))">true</ImportWindowsDesktopTargets>
    <ImportWindowsDesktopTargets Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">true</ImportWindowsDesktopTargets>
  </PropertyGroup>
```